tools/sds-aspect-model-maven-plugin/src/main/java/io/openmanufacturing/sds/aspectmodel/AspectModelMojo.java

```diff
@@ -0,0 +1,151 @@
+            new AspectModelResolver().resolveAspectModel( new FileSystemStrategy( modelsRoot ), urn ) );
+   }
+
+   protected VersionedModel loadModelOrFail( final String aspectModelFilePath ) throws MojoExecutionException {
+      final Try<VersionedModel> versionedModel = loadAndResolveModel( aspectModelFilePath );
+      if ( versionedModel.isFailure() ) {
+         final Throwable loadModelFailureCause = versionedModel.getCause();
+
+         // Model can not be loaded, root cause e.g. File not found
+         if ( loadModelFailureCause instanceof IllegalArgumentException ) {
+            final String errorMessage = String.format( "Can not open file for reading: %s.", aspectModelFilePath );
+            throw new MojoExecutionException( errorMessage );
+         }
+
+         if ( loadModelFailureCause instanceof ModelResolutionException ) {
+            throw new MojoExecutionException( "Could not resolve all model elements" );
+         }
+
+         // Another exception, e.g. syntax error. Let the validator handle this
+         final AspectModelValidator validator = new AspectModelValidator();
+         final ValidationReport report = validator.validate( versionedModel );
+         throw new MojoExecutionException( report.toString() );
+      }
+      return versionedModel.get();
+   }
+
+   protected VersionedModel loadButNotResolveModel() throws MojoExecutionException {
+      final File inputFile = new File( aspectModelFilePath ).getAbsoluteFile();
+      try ( final InputStream inputStream = new FileInputStream( inputFile ) ) {
+         final SdsAspectMetaModelResourceResolver metaModelResourceResolver = new SdsAspectMetaModelResourceResolver();
+         final Try<VersionedModel> versionedModel = TurtleLoader.loadTurtle( inputStream )
+               .flatMap( model -> metaModelResourceResolver.getBammVersion( model )
+               .flatMap( metaModelVersion -> metaModelResourceResolver.mergeMetaModelIntoRawModel( model, metaModelVersion ) ) );
+         if ( versionedModel.isFailure() ) {
+            throw new MojoExecutionException( "Failed to load Aspect model.", versionedModel.getCause() );
+         }
+         return versionedModel.get();
+      } catch ( final IOException exception ) {
+         throw new MojoExecutionException( "Failed to load Aspect model.", exception );
+      }
+   }
+
+   protected FileOutputStream getStreamForFile( final String artifactName, final String outputDirectory ) {
+      try {
+         final File directory = new File( outputDirectory );
+         directory.mkdirs();
+         final File file = new File( directory.getPath() + File.separator + artifactName );
+         return new FileOutputStream( file );
+      } catch ( final FileNotFoundException exception ) {
+         throw new RuntimeException( "Output file for Aspect model documentation generation not found.", exception );
+      }
+   }
+
+   protected AspectModelUrn fileToUrn( final File inputFile ) throws MojoExecutionException {
```

Comment:
Now that we have Code like this duplicated from the bamm-cli, maybe this should be refactored into a new shared module, e.g. `sds-aspect-model-filesystem`?